src/utils/style-utils.js

```diff
@@ -91,17 +91,20 @@ function updateStyleSource(map, update) {
     if (oldSource.type === 'geojson') {
       // update data if no other GeoJSONSource options were changed
       const oldOpts = oldSource.workerOptions;
+      // https://github.com/mapbox/mapbox-gl-js/blob/master/src/source/geojson_source.js
```

Comment:
Nit: Might be worth to explain in a few words why this is needed (in addition to the link). I clicked on it but not immediately obvious.